### ISSUE TITLE
"user cancelled certificate check" when trying to create new pull request

### ISSUE THREAD

**Issue by nk111** [github/VisualStudio#1532]
- GitHub Extension for Visual Studio version: v2.4.3.1737
- Visual Studio version: Visual Studio 2017 (15.6.27428.2002)

I'm using the github for visual studio extension in VS2017. When i try to create a new pull request i always get the error message "user cancelled certificate check".

This is the log:
2018-03-13 17:35:10.332 INFO [56] GitHubPackage Initializing GitHub Extension v2.4.3.1737 in Microsoft Visual Studio 2017 (15.6.27428.2002)
2018-03-13 17:35:32.470 EROR [12] PullRequestCreationViewModel Error creating pull request
LibGit2Sharp.LibGit2SharpException: user cancelled certificate check:
at LibGit2Sharp.Core.Ensure.HandleError(Int32 result)
at LibGit2Sharp.Core.Proxy.git_remote_push(RemoteHandle remote, IEnumerable1 refSpecs, GitPushOptions opts) at LibGit2Sharp.Network.Push(Remote remote, IEnumerable1 pushRefSpecs, PushOptions pushOptions)
at LibGit2Sharp.Network.Push(Remote remote, String pushRefSpec, PushOptions pushOptions)
at LibGit2Sharp.Network.Push(Remote remote, String objectish, String destinationSpec, PushOptions pushOptions)
at GitHub.Services.GitClient.<>c__DisplayClass7_0.b__0() in C:\projects\visualstudio\src\GitHub.App\Services\GitClient.cs:line 64
at System.Threading.Tasks.Task.InnerInvoke()
at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at GitHub.Services.PullRequestService.d__42.MoveNext() in C:\projects\visualstudio\src\GitHub.App\Services\PullRequestService.cs:line 659

Any ideas how to fix this?

**Comment by jcansdale** [comment]
@niik,

Thanks for the information. This is the second time we've had a report like this. I'm not sure what causes it yet.

Here is someone who was seeing a similar exception:
https://mengyiyuanblog.wordpress.com/2016/11/29/configure-bower-with-company-proxy/

Are you working behind a company proxy? If you are, could you try again using a different network?

**Comment by jcansdale** [comment]
@nk111 are you by any chance using Window 7?

**Comment by nk111** [comment]
@jcansdale : yes this happens on a win7 machine. And no, we are not using a proxy (at least no explicit proxy). 

Let me know when i can contribute any more details.

**Comment by trefrym** [comment]
I'm also receiving this error when trying to create a new Pull Request in Visual Studio 2015 on Windows 7

**Comment by johmarjac** [comment]
I also receive it on Win7 VS2017

**Comment by tracstarr-ua** [comment]
The same issue on Win7, VS2017, no proxy.

**Comment by RoLYroLLs** [comment]
I know this is VS2017 specific, but it maybe related to Git. I get this same error when using RedGate's SQL Source Control and `Push` or `Pull` from within their UI. It used to work, I'm not sure if it stopped working due to an update to VS2017 I made a while back.

**Comment by ricksladkey** [comment]
I had the same error and I found a fix.

A comment I read [here](https://github.com/JuliaLang/julia/issues/26167) was the secret. I used the "Easy Fix" method described in this Microsoft support article: [Update to enable TLS 1.1 and TLS 1.2 as a default secure protocols in WinHTTP in Windows](https://support.microsoft.com/en-us/help/3140245/update-to-enable-tls-1-1-and-tls-1-2-as-a-default-secure-protocols-in).
